crates/hir-def/src/lang_item.rs

```diff
@@ -2,101 +2,36 @@
 use intern::{Symbol, sym};
-use rustc_hash::FxHashMap;
+use stdx::impl_from;
 
 use crate::{
     AdtId, AssocItemId, AttrDefId, Crate, EnumId, EnumVariantId, FunctionId, ImplId, ModuleDefId,
     StaticId, StructId, TraitId, TypeAliasId, UnionId,
     db::DefDatabase,
-    expr_store::path::Path,
     nameres::{assoc::TraitItems, crate_def_map, crate_local_def_map},
 };
 
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
 pub enum LangItemTarget {
     EnumId(EnumId),
-    Function(FunctionId),
-    ImplDef(ImplId),
-    Static(StaticId),
-    Struct(StructId),
-    Union(UnionId),
-    TypeAlias(TypeAliasId),
-    Trait(TraitId),
-    EnumVariant(EnumVariantId),
+    FunctionId(FunctionId),
+    ImplId(ImplId),
+    StaticId(StaticId),
+    StructId(StructId),
+    UnionId(UnionId),
+    TypeAliasId(TypeAliasId),
+    TraitId(TraitId),
+    EnumVariantId(EnumVariantId),
 }
 
-impl LangItemTarget {
-    pub fn as_enum(self) -> Option<EnumId> {
-        match self {
-            LangItemTarget::EnumId(id) => Some(id),
-            _ => None,
-        }
-    }
-
-    pub fn as_function(self) -> Option<FunctionId> {
-        match self {
-            LangItemTarget::Function(id) => Some(id),
-            _ => None,
-        }
-    }
-
-    pub fn as_impl_def(self) -> Option<ImplId> {
-        match self {
-            LangItemTarget::ImplDef(id) => Some(id),
-            _ => None,
-        }
-    }
-
-    pub fn as_static(self) -> Option<StaticId> {
-        match self {
-            LangItemTarget::Static(id) => Some(id),
-            _ => None,
-        }
-    }
-
-    pub fn as_struct(self) -> Option<StructId> {
-        match self {
-            LangItemTarget::Struct(id) => Some(id),
-            _ => None,
-        }
-    }
-
-    pub fn as_trait(self) -> Option<TraitId> {
-        match self {
-            LangItemTarget::Trait(id) => Some(id),
-            _ => None,
-        }
-    }
-
-    pub fn as_enum_variant(self) -> Option<EnumVariantId> {
-        match self {
-            LangItemTarget::EnumVariant(id) => Some(id),
-            _ => None,
-        }
-    }
-
-    pub fn as_type_alias(self) -> Option<TypeAliasId> {
-        match self {
-            LangItemTarget::TypeAlias(id) => Some(id),
-            _ => None,
-        }
-    }
-
-    pub fn as_adt(self) -> Option<AdtId> {
-        match self {
-            LangItemTarget::Union(it) => Some(it.into()),
-            LangItemTarget::EnumId(it) => Some(it.into()),
-            LangItemTarget::Struct(it) => Some(it.into()),
-            _ => None,
-        }
-    }
-}
+impl_from!(
+    EnumId, FunctionId, ImplId, StaticId, StructId, UnionId, TypeAliasId, TraitId, EnumVariantId for LangItemTarget
+);
 
 /// Salsa query. This will look for lang items in a specific crate.
 #[salsa_macros::tracked(returns(ref))]
-pub fn crate_lang_items(db: &dyn DefDatabase, krate: Crate) -> Option<Box<LangItems>> {
+pub fn crate_lang_items(db: &dyn DefDatabase, krate: Crate) -> LangItems {
```

Comment:
I think we should keep the `Option<Box<_>>` here, 99.9% of all crates will have this by completely empty after all, unlike the `lang_items` query